scipy/optimize/_linprog_highs.py

```diff
@@ -385,27 +423,24 @@ def _linprog_highs(lp, solver, time_limit=None, presolve=True,
     # HiGHS represents constraints as lhs/rhs, so
     # Ax + s = b => Ax = b - s
     # and we need to split up s by A_ub and A_eq
-    slack = res.get('slack', None)
-    con = None
-    if slack is not None:
-        con = slack[len(b_ub):]
-        slack = slack[:len(b_ub)]
+    if res.get('slack'):
+        slack = res.get('slack')
+        con = np.array(slack[len(b_ub):])
+        slack = np.array(slack[:len(b_ub)])
+    else:
+        slack, con = None, None
 
-    sol = {'x': res.get('x', None),
+    sol = {'x': np.array(res['x']) if res.get('x') else None,
            'slack': slack,
            # TODO: Add/test dual info like:
-           # 'lambda': res.get('lambda', None),
-           # 's': res.get('s', None),
-           'fun': res.get('fun', None),
+           # 'lambda': res.get('lambda'),
+           # 's': res.get('s'),
+           'fun': res.get('fun'),
            'con': con,
            'status': statuses[res['status']][0],
            'success': res['status'] == MODEL_STATUS_OPTIMAL,
            'message': statuses[res['status']][1],
            'nit': res.get('simplex_nit', 0) or res.get('ipm_nit', 0),
            'crossover_nit': res.get('crossover_nit'),
            }
-    if sol['x'] is not None:
```

Comment:
Do you think this is cleaner?

Reply:
I do like it better without the `if` block, personally